qute.jdt/com.redhat.qute.jdt/src/main/java/com/redhat/qute/commons/datamodel/resolvers/SourceTypeKind.java

```diff
@@ -0,0 +1,32 @@
+/*******************************************************************************
+* Copyright (c) 2022 Red Hat Inc. and others.
+* All rights reserved. This program and the accompanying materials
+* which accompanies this distribution, and is available at
+* http://www.eclipse.org/legal/epl-v20.html
+*
+* SPDX-License-Identifier: EPL-2.0
+*
+* Contributors:
+*     Red Hat Inc. - initial API and implementation
+*******************************************************************************/
+package com.redhat.qute.commons.datamodel.resolvers;
+
+/**
+ * Represents the kind of the source type for a piece of value information.
+ * 
+ * For instance, a class annotated with {@code TemplateExtension} can generate
+ * values that can be accessed like fields in a Qute template. Therefore, this
+ * class is a value resolver source type. The "kind" of this value resolver
+ * source type is TemplateExtension, since it is annotated with
+ * {@code @TemplateExtension}. This helps to distinguish it from other classes
+ * that create values that can be accessed like fields in a Qute template.
+ * 
+ * @author datho7561
+ */
+public enum SourceTypeKind {
+	TemplateExtension;
```

Comment:
Could you please support other kinds like TemolateEnum, TemplateGlobal etc

Reply:
TemolateExtension can be annotated to à method and not only for source type.  I think you should rename this enum with ValueResolverKind for instance